src/Core/ComponentValidator.vala

```diff
@@ -0,0 +1,65 @@
+// -*- Mode: vala; indent-tabs-mode: nil; tab-width: 4 -*-
+/*-
+ * Copyright (c) 2017 elementary LLC. (https://elementary.io)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Authored by: Adam Bieńkowski <donadigos159@gmail.com>
+ */
+
+public class AppCenterCore.ComponentValidator : Object {
+    private string[] blacklist;
+
+    private static GLib.Once<ComponentValidator> instance;
+    public static unowned ComponentValidator get_default () {
+        return instance.once (() => { return new ComponentValidator (); });
+    }
+
+    private ComponentValidator () {
+        string blacklist_path = Path.build_filename (Build.CONFIGDIR, "appcenter.blacklist");
+
+        try {
+            string contents;
+            FileUtils.get_contents (blacklist_path, out contents);
+            parse_and_populate (contents);
+        } catch (FileError e) {
```

Comment:
No, it won't crash. You can easily verify that: there is nothing added to `blacklist` currently because there are no entries in the blacklist file. The `blacklist` by default is an empty array.